pkg/apis/resource/validation/validation.go

```diff
@@ -592,23 +594,61 @@ func validateResourceSliceSpec(spec, oldSpec *resource.ResourceSliceSpec, fldPat
 	if spec.AllNodes {
 		setFields = append(setFields, "`allNodes`")
 	}
+	if spec.PerDeviceNodeSelection {
+		setFields = append(setFields, "`perDeviceNodeSelection`")
+	}
 	switch len(setFields) {
 	case 0:
-		allErrs = append(allErrs, field.Required(fldPath, "exactly one of `nodeName`, `nodeSelector`, or `allNodes` is required"))
+		allErrs = append(allErrs, field.Required(fldPath, "exactly one of `nodeName`, `nodeSelector`, `allNodes`, `perDeviceNodeSelection` is required"))
```

Comment:
is required, but none are set

Reply:
The Required error type already indicate the message "none are set" and it aligns with other required API validation pattern. Let me know if you feel super strong on this

Reply:
Yeah, usually "Required" does not use the extra argument.  The main exception is one-of cases andf I think "none are set" is usually implied